src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,17 +57,17 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request: Publish => {
```

Comment:
This is like a typecast, so the use of the request@Publish(_,...,_) pattern is much cleaner